### PR TITLE
feat(use-hooks): create @scalar/use-hooks with useColorMode and useClipboard

### PR DESCRIPTION
**Pull request by hwkr**
Hooks! Tried to clean them up and add some tests best I could ✨

### CODE REVIEW

**Review by hanspagel — DISMISSED** [review]
I like the idea! I just skimmed through the files and left a few suggestions, didn’t test the PR (yet).

File: packages/api-client/src/layouts/Web/ApiClientWeb.vue [hunk 1]
```diff
@@ -6,13 +6,19 @@ import { type HotKeyEvent, handleHotKeyDown } from '@/libs'
...
 import { computed, onBeforeMount, onBeforeUnmount, onMounted } from 'vue'
 import { RouterView } from 'vue-router'
 
 // Initialize dark mode state globally
 useDarkModeState()
 
+// Test the useHello and useGoodbye hooks
+useHello()
+useGoodbye()
+
```
[the remark this answers is not in the document]
Comment (hwkr) [reply packages/api-client/src/layouts/Web/ApiClientWeb.vue]: Yeah that shouldn't be there anymore 😅

**Review by amritk — APPROVED** [review]
LGTM! did not test

**Review by geoffgscott — APPROVED** [review]

File: packages/use-hooks/package.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  "license": "MIT",
+  "author": "Scalar (https://github.com/scalar)",
+  "homepage": "https://github.com/scalar/scalar",
+  "bugs": "https://github.com/scalar/scalar/issues/new/choose",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/scalar/scalar.git",
+    "directory": "packages/use-hooks"
+  },
+  "version": "0.0.0",
```
Comment (marclave) [inline packages/use-hooks/package.json:13]: needs a version :)

**Review by marclave — APPROVED** [review]

**Comment by marclave** [comment]
not tested but needs a version! then can release

File: packages/use-hooks/package.json (hunk 2 above)
Comment (hwkr) [reply packages/use-hooks/package.json:13]: 😅 https://github.com/scalar/scalar/pull/3868#discussion_r1839789784

**Pull request merged**